tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -0,0 +1,278 @@
+from pyanaconda.modules.subscription.subscription import SubscriptionService
+from pyanaconda.modules.subscription.subscription_interface import SubscriptionInterface
+from pyanaconda.modules.subscription.system_purpose import get_valid_fields, normalize_field, \
+    match_field
+from tests.nosetests.pyanaconda_tests import check_kickstart_interface
+
+# content of a valid populated valid values json file for system purpose testing
+SYSPURPOSE_VALID_VALUES_JSON = """
+{
+"role" : ["role_a", "role_b", "role_c"],
+"service_level_agreement" : ["sla_a", "sla_b", "sla_c"],
+"usage" : ["usage_a", "usage_b", "usage_c"]
+}
+"""
+
+# content of a valid but not populated valid values json file for system purpose testing
+SYSPURPOSE_VALID_VALUES_JSON_EMPTY = """
+{
+"role" : [],
+"service_level_agreement" : [],
+"usage" : []
+}
+"""
+
+
+class SystemPurposeLibraryTestCase(unittest.TestCase):
+    """Test the system purpose data handling code."""
+
+    def system_purpose_valid_json_parsing_test(self):
+        """Test that the JSON file holding valid system purpose values is parsed correctly."""
+        # check file missing completely
+        # - use path in tempdir to a file that has not been created and thus does not exist
+        with tempfile.TemporaryDirectory() as tempdir:
+            no_file = os.path.join(tempdir, "foo.json")
+            roles, slas, usage_types = get_valid_fields(valid_fields_file_path=no_file)
+            self.assertListEqual(roles, [])
+            self.assertListEqual(slas, [])
+            self.assertListEqual(usage_types, [])
+
+        # check empty value list is handled correctly
+        with tempfile.NamedTemporaryFile(mode="w+t") as testfile:
+            testfile.write(SYSPURPOSE_VALID_VALUES_JSON_EMPTY)
+            testfile.flush()
+            roles, slas, usage_types = get_valid_fields(valid_fields_file_path=testfile.name)
+            self.assertListEqual(roles, [])
+            self.assertListEqual(slas, [])
+            self.assertListEqual(usage_types, [])
+
+        # check correctly populated json file is parsed correctly
+        with tempfile.NamedTemporaryFile(mode="w+t") as testfile:
+            testfile.write(SYSPURPOSE_VALID_VALUES_JSON)
+            testfile.flush()
+            roles, slas, usage_types = get_valid_fields(valid_fields_file_path=testfile.name)
+            self.assertListEqual(roles, ["role_a", "role_b", "role_c"])
+            self.assertListEqual(slas, ["sla_a", "sla_b", "sla_c"])
+            self.assertListEqual(usage_types, ["usage_a", "usage_b", "usage_c"])
+
+    def normalize_field_test(self):
+        """Test that the system purpose valid field normalization works."""
+        # this should basically just lower case the input
+        self.assertEqual(normalize_field("AAA"), "aaa")
+        self.assertEqual(normalize_field("Ab"), "ab")
+        self.assertEqual(normalize_field("A b C"), "a b c")
```

Comment:
Do we need this test? Isn't this already tested by the above test?

Reply:
I tihnk it is good to have all the `system_purpose` library field processing functions covered.